geopandas/io/tests/test_file.py

```diff
@@ -694,6 +695,17 @@ def test_allow_legacy_gdal_path(engine, nybb_filename):
     assert isinstance(gdf, geopandas.GeoDataFrame)
 
 
+def test_read_file_with_hash_in_path(engine, nybb_filename, tmp_path):
+    if engine == "pyogrio":
+        pytest.xfail("upstream bug")
```

Comment:
we can turn this into a conditional skip once https://github.com/geopandas/pyogrio/pull/412 is in.

Reply:
Can be updated now